core/src/avm2/e4x.rs

```diff
@@ -1249,17 +1267,33 @@ pub fn maybe_escape_child<'gc>(
         if child.as_object().map_or(false, |x| {
             x.as_xml_object().is_some() || x.as_xml_list_object().is_some()
         }) {
-            Ok(child)
+            return Ok(child);
         } else {
             let string = child.coerce_to_string(activation)?;
             let xml = activation
                 .avm2()
                 .classes()
                 .xml
                 .construct(activation, &[string.into()])?;
-            Ok(xml.into())
+            return Ok(xml.into());
         }
-    } else {
-        Ok(child)
     }
+
+    // FIXME: This should use caller movie version.
+    if activation.context.swf.version() >= 21 {
```

Comment:
`activation.caller_movie().unwrap().version()`

Reply:
Originally used that, but some tests were panicking since caller_movie was missing?

Reply:
I'm PRing a fix right now

Reply:
Tests should no longer fail if you rebase